git_dumper.py

```diff
@@ -644,7 +671,11 @@ def fetch_git(url, directory, jobs, retry, timeout, http_headers, client_cert_p1
     os.chdir(directory)
 
     # ignore errors
-    subprocess.call(["git", "checkout", "."], stderr=open(os.devnull, "wb"))
+    subprocess.call(
+        ["git", "checkout", "."], 
+        stderr=open(os.devnull, "wb"),
+        env=ENV
+    )
```

Comment:
I think `ENV` is only created within that `if` line 469 so it wouldn't exist here (since that `if` returns).
We also need to sanitize the .git/config here.
We could move the whole git checkout step in a function to avoid code duplication.

Reply:
This comment wasn't addressed.
Notice how the if line 450 implements a different stategy where we just download recursively using directory indexing, but if that if is not taken, we use a different strategy. In that case, you are not sanitizing the .git/config and you are not setting up the `environment` variable either.
We need two changes:
* Move the `environment` variable creation and setup before the if line 450
* Add a `Sanitizing .git/config` step here (line 647) before calling `git checkout .`